homie5/values/boolean.yml

```
@@ -2,69 +2,98 @@ description: validating boolean values
 
 tests:
   - description: Passes valid boolean values
-    type: boolean
-    value: 'true'
+    type: propertyvalue
+    definition:
+      datatype: boolean
+    input_data: "true"
     valid: true
   - description: Passes valid boolean values
-    type: boolean
-    value: 'false'
+    type: propertyvalue
+    definition:
+      datatype: boolean
+    input_data: "false"
     valid: true
   - description: an empty string is not valid
-    type: boolean
-    value: ''
+    type: propertyvalue
+    definition:
+      datatype: boolean
+    input_data: ""
```

Should the test case have this?
```suggestion
    input_data: "\x00"
```

---

I just used the test file you provided for now. I did not do an evaluation of the actual tests but focused on the data format for the tests first. What is the reason for using a string with a 0 byte value?

---

specs says empty strings should be encoded as a 0-byte. By doing it like this, it will ensure that this edge case is taken into account.

---

Ah right sorry. Makes sense